modules/tinymce/src/plugins/visualchars/main/ts/core/Nodes.ts

```diff
@@ -11,17 +11,32 @@ const isMatch = (n: SugarElement<Node>): n is SugarElement<Text> => {
     Data.regExp.test(value);
 };
 
-// inlined sugars PredicateFilter.descendants for file size
-const filterDescendants = <T extends Node>(scope: SugarElement<Node>, predicate: (x: SugarElement<Node>) => x is SugarElement<T>): SugarElement<T>[] => {
+const isContentEditableFalse = (node: SugarElement<Node>) => SugarNode.isHTMLElement(node) && ContentEditable.getRaw(node) === 'false';
+
+const isChildEditable = (node: SugarElement<Node>, currentState: boolean) => {
+  if (SugarNode.isHTMLElement(node)) {
+    const value = ContentEditable.getRaw(node);
+    if (value === 'true') {
+      return true;
+    } else if (value === 'false') {
+      return false;
+    }
+  }
```

Comment:
The problem is that the isEditable does a closest check and since we are walking the dom as a tree level by level I only need to check if the state has changed when I traverse down. So it becomes a O(1) instead of a O(n) on the depth of the tree. The code refactor you suggest is smaller but does have potential performance implications.